foojay-resolver/src/main/kotlin/org/gradle/toolchains/foojay/distributions.kt

```diff
@@ -30,24 +30,34 @@ fun match(
         vendor: JvmVendorSpec,
         implementation: JvmImplementation,
         version: JavaLanguageVersion
-): Distribution? {
-    if (JvmImplementation.J9 == implementation) {
-        return distributions.firstOrNull { it.name == j9Aliases[vendor] }
-    }
-    if (JvmVendorSpec.GRAAL_VM == vendor) {
-        return match(distributions, JvmVendorSpec.matching("Graal VM CE " + version.asInt()))
+): List<Distribution> = when {
+    JvmImplementation.J9 == implementation -> distributions.filter { it.name == j9Aliases[vendor] }
```

Comment:
Incorrect. In the spirit of the changes we are making vendor=any(), implementation=J9 should return multiple distributions, but it doesn't.